utopia-remix/app/routes/v1.project.$id.tsx

```diff
@@ -1,14 +1,40 @@
 import { ActionFunctionArgs, LoaderFunctionArgs } from '@remix-run/node'
 import { proxy } from '../util/proxy.server'
-import { handle, handleOptions } from '../util/api.server'
+import { ensure, handle, handleOptions, requireUser } from '../util/api.server'
+import { Params } from '@remix-run/react'
+import { Status } from '../util/statusCodes'
+import { hasUserProjectPermission } from '../services/permissionsService.server'
+import { UserProjectPermission } from '../types'
 
 export async function loader(args: LoaderFunctionArgs) {
   return handle(args, {
     OPTIONS: handleOptions,
-    GET: proxy,
+    GET: getProject,
   })
 }
 
+export async function getProject(req: Request, params: Params<string>) {
+  const { id: projectId } = params
+  ensure(projectId != null, 'project is null', Status.BAD_REQUEST)
+  let userId
+  try {
+    const user = await requireUser(req)
+    userId = user.user_id
+  } catch (e) {
+    userId = 'ANON'
+  }
+  const projectData = (await proxy(req, params)) as { ownerId: string }
```

Comment:
I'd drop the proxying here, since we're returning the data directly without streaming it and hence losing the related benefits. You can just grab the data directly from the DB at this point

Reply:
so is it better to just query the DB directly? the reason I chose to proxy here is just to allow for any logic the Haskell server does that the BFF is not (and should not be) aware of

Reply:
I just felt with the code involved here the proxying felt a bit unnecessary, but if going with the middleware solution I'd indeed just keep the proxying there for now (I'd lean on the latter)